File: code base/1-jpa-demo/src/main/java/com/cg/entity/Customer.java
package com.cg.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity //maps this class with the Customer table
 //maps all instance variables with table cols
@Table(name="CUSTOMER_DATA") //optional (default table name is class name)
public class Customer {
	@Id//for PK (its mandatory)
 private int custId;
	@Column(name="cname",length = 50) //optional
 private String custName;
 private String address;
 @Column(name="contactNo",unique = true)
 private int phoneNo;
 
public int getPhoneNo() {
	return phoneNo;
}
public void setPhoneNo(int phoneNo) {
	this.phoneNo = phoneNo;
}
public int getCustId() {
	return custId;
}
public void setCustId(int custId) {
	this.custId = custId;
}
public String getCustName() {
	return custName;
}
public void setCustName(String custName) {
	this.custName = custName;
}
public String getAddress() {
	return address;
}
public void setAddress(String address) {
	this.address = address;
}
 
 
 
}

File: FrontEnd/html/Typescript/Customer.ts
export class Customer{
    cid:number; //default access specifier is public (private, public and protected)
    cname:string;
    address:string;
   
    constructor (cid:number,cname:string, address:string){
        this.cid=cid;
        this.cname=cname;
        this.address=address;
    }
    public getData():string{
        return `customer id is ${this.cid} 
         and name is ${this.cname} and address is 
         ${this.address}`;
    }
}